### PR TITLE
Support Resource, ResourceType only

### PR DESCRIPTION
**Pull request by ari-becker**
Drops the old, dormant implementation for a repository that focuses on
hosting the types / schemas of the various resource types, and allows
the downstream user to construct the type of their Resource and
ResourceType as necessary. This solves the code-reusage problem while
keeping downstream implementations performant.

### CODE REVIEW

File: ResourceType.dhall [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            , privileged : Optional Bool
+            , params : Optional _params.SourceGetParams
+            , check_every : Optional Text
+            , tags : Optional (List Text)
+            , unique_version_history : Optional Bool
+            }
+        , default =
+          { privileged = None Bool
+          , params = None _params.SourceGetParams
+          , check_every = None Text
```
Comment (oded-dd) [inline ResourceType.dhall:26]: Shouldn't the default be 24h?

Comment (ari-becker) [reply ResourceType.dhall:26]: @oded-dd The upstream field is optional... I can imagine more than a number of cases where we can't use webhooks to trigger checking the resource and as such benefit from a shorter check_every

File: Resource.dhall [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        , default :
+            { icon : Optional Text
+            , version : Optional _params.Version
+            , check_every : Optional Text
+            , tags : Optional (List Text)
+            , public : Optional Bool
+            , webhook_token : Optional Text
+            }
+        }
+    = λ(_params : { Source : Type, Version : Type }) →
```
Comment (oded-dd) [inline Resource.dhall:13]: Why not just getting the `\(resource-types : Tyoe) ->` as anyways `resource-types/*.dhall` holds all the required input.

File: ResourceType.dhall [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        { Type : Type
+        , default :
+            { privileged : Optional Bool
+            , params : Optional _params.SourceGetParams
+            , check_every : Optional Text
+            , tags : Optional (List Text)
+            , unique_version_history : Optional Bool
+            }
+        }
+    = λ(_params : { Source : Type, SourceGetParams : Type }) →
```
Comment (oded-dd) [inline ResourceType.dhall:12]: Why not just getting the `\(resource-types : Tyoe) ->` as anyways `resource-types/*.dhall` holds all the required input.

**Review by oded-dd — CHANGES_REQUESTED** [review]
Unless I missed something I don't see why the function for both Resource and ResourceType will not only get `\(resource-types : Type) ->`

Also I dont see any step implementation, is it safe to assume that this part will be per repo which contains the pipeline?

**Review by oded-dd — APPROVED** [review]

**Pull request merged**